api/v1alpha1/proposedcommit_types.go

```diff
@@ -40,36 +40,38 @@ type ProposedCommitSpec struct {
 	// +kubebuilder:validation:Required
 	ActiveBranch string `json:"activeBranch"`
 
-	CommitStatuses []CommitStatusProposedCommitSpec `json:"commitStatuses,omitempty"`
-}
+	// ActiveCommitStatuses what statuses are required for the active branch
+	// +kubebuilder:validation:Optional
+	ActiveCommitStatuses []CommitStatusSelector `json:"activeCommitStatuses"`
 
-type CommitStatusProposedCommitSpec struct {
-	// Key staging hydrated branch
-	// +kubebuilder:validation:Required
-	Key string `json:"key"`
+	// ProposedCommitStatuses what statuses are required for the proposed branch
+	// +kubebuilder:validation:Optional
+	ProposedCommitStatuses []CommitStatusSelector `json:"proposedCommitStatuses"`
 }
 
-type CommitStatusProposedCommitStatus struct {
+type ProposedCommitCommitStatusState struct {
 	// Key staging hydrated branch
 	// +kubebuilder:validation:Required
 	Key string `json:"key"`
 
 	// Phase what phase is the status in
 	// +kubebuilder:validation:Required
-	Phase string `json:"phase"`
+	Status string `json:"status"`
 }
 
-type ProposedCommitBranchState struct {
-	Dry      ProposedCommitShaState `json:"dry,omitempty"`
-	Hydrated ProposedCommitShaState `json:"hydrated,omitempty"`
+type CommitBranchState struct {
+	Dry      CommitShaState `json:"dry,omitempty"`
+	Hydrated CommitShaState `json:"hydrated,omitempty"`
+	// +kubebuilder:validation:Optional
+	CommitStatuses []ProposedCommitCommitStatusState `json:"commitStatuses,omitempty"`
```

Comment:
Just confirming if this is the intended name or a typo: `ProposedCommitCommitStatusState`

Reply:
It is intended for lack of a better name, it is the ProposedCommit controllers CommitStatusState resource it conflicts with CommitStatus [struct](https://github.com/argoproj-labs/gitops-promoter/blob/main/api/v1alpha1/commitstatus_types.go#L90) hence the name.

Reply:
It sounds a bit confusing at first. Why not just `ProposedCommitStatusState`?
Feel free to keep it this way if you think this is preferable.

Reply:
thats even more confusing me and Michael did talk a bit about not loving the name of the ProposedCommit CRD but again struggled to find another reasonable name.

The field here is really the CommitStatusState but for the ProposedCommit Controller hence ProposedCommitCommitStatusState

ProposedCommitStatusState to me is confusing because what is it the state of. It is not the state of the ProposedCommit. It is the CommitStatusState of all child CommitStatues.

Naming is hard :(

Reply:
I will probably leave as is for now we might re-work some names of CRDs etc etc in the future as well, the API names though are what really matter and I think we are pretty happy with them.